### PR TITLE
build: Add pre-commit and Black

### PR DESCRIPTION
**Pull request by dcramer**

### CODE REVIEW

**Comment by dcramer** [comment]
Not wired up to travis since you're running docker builds. Could add it fairly easily tho.

**Comment by dcramer** [comment]
Also needs a pass at Black to "fix" things. ``pre-commit run -a``

**Comment by dcramer** [comment]
Also not running flake8 in CI right now, probably worth fixing that too, but pre-commit would handle both.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+exclude: "node_modules$"
```
Comment (fpacifici) [inline .pre-commit-config.yaml]: there are no node modules in Snuba.

**Review by fpacifici — APPROVED** [review]
Looking at the outstanding PRs nothing will be a nightmare to rebase/merge, so we can go ahead.
Could you please remove the node_modules line before committing ?

**Comment by dcramer** [comment]
Yep will remove node_modules. Want me to do a formatting pass for this as well?

**Comment by fpacifici** [comment]
_Yep will remove node_modules. Want me to do a formatting pass for this as well?_
Sure, thanks. Seems a good time looking at the outstanding PRs

**Comment by dcramer** [comment]
removing pyflakes and pycodestyle from reqs, should fix the build

**Pull request merged**